### ISSUE TITLE
augmenting speech data with rnnoise Dataset

### ISSUE THREAD

**Issue by alokprasad** [mozilla/DeepSpeech#2519]
For support and discussions, please use our [Discourse forums](https://discourse.mozilla.org/c/deep-speech).

If you've found a bug, or have a feature request, then please create an issue with the following information:

- **Have I written custom code (as opposed to running examples on an unmodified clone of the repository)**: ds 0.6 alpha 11 patch with augementing rrnoise dataset
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 18.04
- **TensorFlow installed from (our builds, or upstream TensorFlow)**: 1.14
- **TensorFlow version (use command below)**:
- **Python version**: 3.6.8
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

You can obtain the TensorFlow version with

```bash
python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"
```

I am trying to augement speech dataset with https://people.xiph.org/~jm/demo/rnnoise/rnnoise_contributions.tar.gz 
it code randomly picks one noise sample ( i have deleted files which are smaller than speech file)

with below patch .
```
root@1b9f54a4e989:/home/alok# diff -up DeepSpeech/util/feeding.py DeepSpeech/util/feeding_rnnoise.py 
--- DeepSpeech/util/feeding.py  2019-11-11 04:31:15.134872546 +0000
+++ DeepSpeech/util/feeding_rnnoise.py  2019-11-11 04:30:35.824470109 +0000
@@ -65,11 +65,24 @@ def samples_to_mfccs(samples, sample_rat
 
+def audiofile_to_features(speech_file, train_phase=False):
+    import os,random
+    #delete some files from rnnoise_contributions less than maximum size or added silence or noise 
+    noise_file = random.choice(os.listdir("/mnt/ds_dataset/rnnoise_contributions/wavs-16k"))
+    noise_file = os.path.join("/mnt/ds_dataset/rnnoise_contributions/wavs-16k",noise_file)
+    samples_speech = tf.io.read_file(speech_file)
+    decoded_speech = contrib_audio.decode_wav(samples_speech, desired_channels=1)
+
+    samples_noise = tf.io.read_file(noise_file)
+    decoded_noise = contrib_audio.decode_wav(samples_noise, desired_channels=1)
+
+    #TODO://trim noise file to size of speech file
+    x = decoded_noise[0][0:tf.shape(decoded_speech[0])[0]]
+    
+    mixed_audio = tf.math.add(decoded_speech[0],x)
+    #final_pcm = contrib_audio.encode_wav(mixed_audio,16000)
 
-def audiofile_to_features(wav_filename, train_phase=False):
-    samples = tf.io.read_file(wav_filename)
-    decoded = contrib_audio.decode_wav(samples, desired_channels=1)
-    features, features_len = samples_to_mfccs(decoded.audio, decoded.sample_rate, train_phase=train_phase)
+    features, features_len = samples_to_mfccs(mixed_audio, decoded_speech.sample_rate, train_phase=train_phase)
 

+#def audiofile_to_features(wav_filename, train_phase=False):
+#    samples = tf.io.read_file(wav_filename)
+#    decoded = contrib_audio.decode_wav(samples, desired_channels=1)
+#    features, features_len = samples_to_mfccs(decoded.audio, decoded.sample_rate, train_phase=train_phase)
+
+ #   if train_phase:
+ #       if FLAGS.data_aug_features_multiplicative > 0:
+ #           features = features*tf.random.normal(mean=1, stddev=FLAGS.data_aug_features_multiplicative, shape=tf.shape(features))
+
+  #      if FLAGS.data_aug_features_additive > 0:
+  #          features = features+tf.random.normal(mean=0.0, stddev=FLAGS.data_aug_features_additive, shape=tf.shape(features))
+
+  # return features, features_len
```
But even validation loss after some epocs is validation loss is quite high with loss less than 1.
anything else has to be done to make it better?

**Comment by alokprasad** [comment]
Dataset is quite huge around 15Gb on extraction , i am augmenting each file..should i augment only few files ? or reduce gain of before mixing with speech.

**Comment by kdavis-mozilla** [comment]
We'd like to reserve issues for bugs in the github code base or feature requests. As this is related to your modified code and a new data set and not related to a feature request or bug, could you move this to [Discourse](https://discourse.mozilla.org/c/deep-speech) where we tackle such issues. Thanks!

**Comment by alokprasad** [comment]
@kdavis-mozilla it could be a feature request to integrate rnnoise dataset to augment with speech or should i create a separate bug for feature request?

**Comment by kdavis-mozilla** [comment]
@alokprasad Have you tried [voice-corpus-tool](https://github.com/mozilla/voice-corpus-tool)? Some [Discourse](https://discourse.mozilla.org/c/deep-speech) comments on it are [here](https://discourse.mozilla.org/t/how-to-use-voice-corpus-tool-to-augment-the-dataset/43250).

**Comment by alokprasad** [comment]
@kdavis-mozilla i just now looked at  voice-corpus-tool but its more of offline augmentation , 
adding the real noise ( not guassian or random noise ) online during training is what i am looking.

**Comment by kdavis-mozilla** [comment]
We've [added](https://github.com/mozilla/DeepSpeech/blob/master/doc/TRAINING.rst#training-with-augmentation) training time augmentation. So I'm closing this for now.
